src/pages/docs/go-w3up.mdx

```diff
@@ -0,0 +1,181 @@
+
+<Steps>
+### Generate a DID
+
+[Generate a DID](#generate-a-did) and make a note of it (the string starting with `did:key:...`)
+
+### Install w3 CLI
+
+```sh
+npm install -g @web3-storage/w3cli
+```
+
+### Create a space
+
+```sh
+w3 space create [NAME]
+```
+
+### Delegate capabilities to your DID
+
+```sh
+w3 delegation create -c 'store/*' -c 'upload/*' [DID] -o proof.ucan
+```
+</Steps>
+
+<Callout>
+  Make a note of the space DID from above. You'll need it later.
+</Callout>
+
+## Load private key and proofs
+
+To interact with the web3.storage API you need your private key to sign UCAN invocations and a proof that your key has been delegated capabilities to perform tasks:
+
+```go
+package main
+
+import (
+  "ioutil"
+
+  "github.com/web3-storage/go-ucanto/did"
+  "github.com/web3-storage/go-ucanto/principal/ed25519/signer"
+  "github.com/web3-storage/go-w3up/delegation"
+)
+
+// space that the client will interact with
+space, _ := did.Parse("did:key:z6MkwDuRThQcyWjqNsK54yKAmzfsiH6BTkASyiucThMtHt1y")
+
+// private key to sign UCAN invocations with
+priv, _ := ioutil.ReadFile("path/to/private.key")
+issuer, _ := signer.Parse(priv)
+
+// UCAN proof(s) that the signer can perform tasks in this space (a delegation chain)
+prfbytes, _ := ioutil.ReadFile("path/to/proof.ucan")
+proof, _ := delegation.ExtractProof(b)
+```
+
+## Upload a CAR
+
+Once you have loaded your space DID, your private key and your delegation proofs, you can upload a CAR to web3.storage.
+
+```go
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	"github.com/ipfs/go-cid"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/multiformats/go-multihash"
+	"github.com/web3-storage/go-w3up/capability/storeadd"
+	"github.com/web3-storage/go-w3up/client"
+)
+
+func main() {
+  data, _ := ioutil.ReadFile("path/to/my.car")
+
+  // generate the CID for the CAR
+  mh, _ := multihash.Sum(data, multihash.SHA2_256, -1)
+  link := cidlink.Link{Cid: cid.NewCidV1(0x0202, mh)}
+
+  rcpt, _ := client.StoreAdd(
+    issuer,
+    space,
+    &storeadd.Caveat{Link: link, Size: len(data)},
+    client.WithProofs(proofs),
+  )
+
+  // "upload" means it needs to be uploaded, "done" means it is already done!
+  if rcpt.Out().Ok().Status == "upload" {
+    hr, _ := http.NewRequest("PUT", *rcpt.Out().Ok().Url, bytes.NewReader(data))
+
+    hdr := map[string][]string{}
+    for k, v := range rcpt.Out().Ok().Headers.Values {
+      hdr[k] = []string{v}
+    }
+
+    hr.Header = hdr
+    hr.ContentLength = len(data)
+    httpClient := http.Client{}
+    res, _ := httpClient.Do(hr)
+    res.Body.Close()
+  }
+
+  fmt.Println(link.String())
+}
+```
+
+<Callout type="warning">
+  Maximum upload size is 4,261,412,864 bytes (around 4GB). To upload CAR files larger than this, please use the [sharding utility](https://github.com/web3-storage/go-w3up/blob/main/car/sharding/sharding.go) to split the CAR into multiple shards.
```

Comment:
```suggestion
  Maximum upload size for a chunk is 4,261,412,864 bytes (around 4GB). To upload CAR files larger than this, please use the [sharding utility](https://github.com/web3-storage/go-w3up/blob/main/car/sharding/sharding.go) to split the CAR into multiple shards.
```

I think it would be good to be explicit as the chunk does not represent what an user is able to upload as part of a dag

Reply:
just read below, and I think it even is nicer then to include chunk detail